contracts/SnappAuction.sol

```diff
@@ -0,0 +1,81 @@
+pragma solidity ^0.5.0;
+
+import "./SnappBase.sol";
+
+
+contract SnappAuction is SnappBase {
+  
+    uint16 public constant AUCTION_BATCH_SIZE = 1000;
+
+    uint public auctionIndex;
+    mapping (uint => PendingFlux) public auctions;
+
+    event SellOrder(
+        uint auctionId, 
+        uint16 slotIndex,
+        uint16 accountId, 
+        uint8 buyToken, 
+        uint8 sellToken, 
+        uint128 buyAmount,
+        uint128 sellAmount
+    );
+
+    constructor () public {
+        auctions[auctionIndex].creationBlock = block.number;
+    }
+
+    function placeSellOrder(
+        uint8 buyToken,
+        uint8 sellToken,
+        uint128 minBuyAmount,
+        uint128 maxSellAmount
+    ) public onlyRegistered() {
+        require(buyToken != sellToken, "Buy and Sell tokens must differ!");
```

Comment:
Interesting. This might be somewhat related to all the requires we found a bit difficult to work with in the dutchX when we tried to build a smart contract that wants to use it.

I agree that it doesn't make sense to place an order in the same currency (just like it doesn't make sense to withdraw 0 tokens), but maybe we should only throw when someone is trying to do something illegal.

Don't need to change, but something to think about/discuss on a more philosophical level.

Reply:
What do you mean by "illegal" here? I was under the impression that "illegal" is defined by the failure to comply with the contract's requirements.

Reply:
I wouldn't consider it illegal to trade GNO for GNO. I would consider it senseless but if someone wants to do it they are not attacking the contract in any malicious/illegal way.

If they found a way to manipulate the orders of other people, that I'd consider an example for illegality that should throw.

Reply:
Yes, this is true. It would make the contract and unit testing a lot simpler if some of these validations were lifted. However, it might not hurt to have all of this in place earlier, and then decide which can be removed. I suspect that the business logic of the external services would then have a lot more conditions to consider and handle. I suppose it all depends where these should be taken care of. I have included them here in the beginning, so to cut the problem off directly at the source. At least they are all recorded somewhere and their handling can be moved around at some point later...

Reply:
I think it is good to allow GNO for GNO orders, as it is more gas efficient. But @twalth3r , your solver need to sort out these orders. Is your solver able to sort these order out or deal with them in any other meaning ful way?

Reply:
So, as I understand, we are waiting on this and then this PR might be ready for merge @fleupold @josojo ?

Reply:
I don't understand why we need to wait for anything here? The solver is not going to be able to match an order where I sell 10 GNO for 11 GNO, because then the price would not be reflexive. It could match orders where we sell 10GNO for 10GNO and if someone wants to do that, I think he should be free to do so.

Reply:
So is there a consensus to remove this constraint from the order validation? Someone just say so and its gone.

Reply:
Forget it. Has been removed. Order are now allowed to have the same token pair.